spec/unit/lib/temporal/connection/converter/payload/json_spec.rb

```diff
@@ -1,17 +1,42 @@
 require 'temporal/connection/converter/payload/json'
 
 describe Temporal::Connection::Converter::Payload::JSON do
-  subject { described_class.new }
+  subject { described_class.new(options) }
+
+  let(:options) { { circular: true } }
 
   describe 'round trip' do
+    let(:expected_oj_options) do
+      {
+        circular: true,
+        float_precision: 0,
+        mode: :object
+      }
+    end
+
     it 'safely handles non-ASCII encodable UTF characters' do
       input = { 'one' => 'one', two: :two, ':three' => '☻' }
 
+      expect(Oj).to receive(:dump)
+        .with(input, expected_oj_options)
+        .and_call_original
+      expect(Oj).to receive(:load)
+        .with(be_a(String), expected_oj_options)
+        .and_call_original
+
       expect(subject.from_payload(subject.to_payload(input))).to eq(input)
     end
 
     it 'handles floats without loss of precision' do
       input = { 'a_float' => 1626122510001.305986623 }
+
+      expect(Oj).to receive(:dump)
+        .with(input, expected_oj_options)
+        .and_call_original
+      expect(Oj).to receive(:load)
+        .with(be_a(String), expected_oj_options)
+        .and_call_original
```

Comment:
Could we test the cases where we need `circular: true` instead of mocking? @Vuta could you help to give an example here?

Reply:
Yes this is one example of the error raised with circular reference

<img width="960" alt="Screenshot 2023-12-14 at 09 56 47" src="https://github.com/Kaligo/temporal-ruby/assets/21086542/64bbb309-560c-476f-a05b-bd2e9bf461ab">

Reply:
Thanks! I refactored the test